src/components/GraphFilter/GraphFilter.tsx

```diff
@@ -47,6 +50,8 @@ export default class GraphFilter extends React.PureComponent<GraphFilterProps> {
   };
 
   render() {
+    const graphTypeKey: string = _.findKey(GraphType, val => val === this.props.graphType)!;
```

Comment:
I tried casting the value. JS does not complain, but at runtime the value is not valid (the test `graphType === GraphType.{SOMETHING}` is always false).

Here, I need the `key` of the enum rather than the enumerated value. I guess this is what makes not possible to type the variable as an enum.

Reply:
typescript enums are not cool: https://basarat.gitbooks.io/typescript/docs/enums.html